cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -108,25 +108,14 @@ func NewCloudControllerManagerOptions() (*CloudControllerManagerOptions, error)
 }
 
 // NewDefaultComponentConfig returns cloud-controller manager configuration object.
-func NewDefaultComponentConfig(insecurePort int32) (*componentconfig.CloudControllerManagerConfiguration, error) {
-	// TODO: This code will be fixed up/improved when the ccm API types are moved to their own, real API group out of
-	// pkg/apis/componentconfig to cmd/cloud-controller-manager/app/apis/
-	scheme := runtime.NewScheme()
-	if err := componentconfigv1alpha1.AddToScheme(scheme); err != nil {
-		return nil, err
-	}
-	if err := componentconfig.AddToScheme(scheme); err != nil {
+func NewDefaultComponentConfig(insecurePort int32) (*ccmconfig.CloudControllerManagerConfiguration, error) {
+	versioned := &ccmconfigv1alpha1.CloudControllerManagerConfiguration{}
+	ccmconfigscheme.Scheme.Default(versioned)
+
+	internal := &ccmconfig.CloudControllerManagerConfiguration{}
+	if err := ccmconfigscheme.Scheme.Convert(versioned, internal, nil); err != nil {
```

Comment:
why delete `*.AddToScheme(scheme)` ?, for  `AddScheme()` in file `cmd/cloud-controller-manager/app/apis/config/scheme/scheme.go`?

Reply:
Yeah earlier this function built its own scheme from scratch, now it's reusing the standardized code in the scheme package which exposes the global `Scheme` variable.